include/openPMD/IO/AbstractIOHandler.hpp

```diff
@@ -131,6 +148,7 @@ class AbstractIOHandler
     virtual std::string backendName() const = 0;
 
     std::string const directory;
+    // why do these need to be separate?
```

Comment:
Do we want to answer this question? :)

I am not sure we documented this well when this was introduced (also visible in the missing doxygen strings for those two members).

Reply:
I ran into that a few times. I believe it would be consistent to treat all files in the file mode as they were one file. When its time to overwrite, remove all existing files related to the  series.

Reply:
> Do we want to answer this question? :)

I'd suggest to do it in a separate PR, need to figure out what the answer is at first though
I wasn't always quite sure which one of these to use when writing this PR

> I ran into that a few times. I believe it would be consistent to treat all files in the file mode as they were one file. When its time to overwrite, remove all existing files related to the series.

We could think about this for CREATE mode, since its semantics are "overwrite anything that exists". The purpose READ_WRITE and APPEND modes is explicitly *not* to delete existing data. For APPEND mode, the most sensible solution seems to be truncation (i.e. "delete anything past iteration 500") which would be a follow-up PR.